Automattic/wp-calypso
# Jetpack Connect: Update cobranded

Before this PR, the co-branded connection screen looked like this:

<img width="300" alt="milesweb-old-demo" src="https://user-images.githubusercontent.com/1126811/39322277-f3be4a72-494e-11e8-9161-375166f8d9df.png">

Notice that there's a bit of jank with the Jetpack logo and the co-branded logo. This is because the Jetpack logo is being render in the `MainWrapper` component while the co-branded logos are rendered further down in the `AuthFormHeader` component.

To address this, I 1) moved the logic for getting the partner slug to the common `utils.js` file and then 2) moved the co-branded logo rendering to `MainWrapper`.

Further, this PR also updates the co-branded logo to use the partner's full logo alongside our logo mark. Part of the impetus for this is that not all partners have a logo mark. To account for that, we now mention Jetpack in the co-branded connection page header.

Here are some screenshots:

<img width="300" alt="screen shot 2018-04-26 at 12 22 46 pm" src="https://user-images.githubusercontent.com/1126811/39322418-7281049e-494f-11e8-89dd-886441045189.png">
<img width="300" alt="screen shot 2018-04-26 at 12 22 56 pm" src="https://user-images.githubusercontent.com/1126811/39322419-729256a4-494f-11e8-99b6-a8b439e7c3e7.png">
<img width="300" alt="screen shot 2018-04-26 at 12 23 03 pm" src="https://user-images.githubusercontent.com/1126811/39322420-72ab8b60-494f-11e8-9003-883d325b6509.png">
<img width="300" alt="screen shot 2018-04-26 at 12 23 11 pm" src="https://user-images.githubusercontent.com/1126811/39322421-72cb662e-494f-11e8-855c-bcf039ff19d1.png">

To test:

- Checkout this branch
- Go to a Jetpack site and click the green "Set up Jetpack" button to connect to WordPress.com
- Once on WordPress.com, copy the connection URL
- Replace WordPress.com with `calypso.localhost:3000`
- To test the various co-branded screens you'll add `&partner_id={$ID}` where `$ID` represents one of the hosts below:
    - `51945` or `51946` for DreamHost
    - `49615` or `49640` for Pressable
    - `57152` or `57733` for MilesWeb
    - `41986` or `42000` for Bluehost
- While going through the various flows, ensure there are no errors
